storage/innobase/lock/lock0lock.cc

```diff
@@ -1842,7 +1842,9 @@ lock_rec_add_to_queue(
 
 		if (lock != NULL) {
 
-			lock_rec_set_nth_bit(lock, heap_no);
+			if (!lock_rec_get_nth_bit(lock, heap_no)) {
```

Comment:
Do we need to same with lock_rec_reset_nth_bit   i.e reset only if it is set?

Reply:
we already do. if bit !=0, it means it is set, only then we decrement.